nncf/experimental/tensor/torch_functions.py

```diff
@@ -37,112 +37,151 @@ def _(a: torch.Tensor) -> TensorDeviceType:
+@fns.dtype.register(torch.Tensor)
 def _(a: torch.Tensor) -> TensorDataType:
     return DTYPE_MAP_REV[a.dtype]
 
 
-@functions.reshape.register(torch.Tensor)
+@fns.reshape.register(torch.Tensor)
 def _(a: torch.Tensor, shape: List[int]) -> torch.Tensor:
     return a.reshape(shape)
 
 
-@functions.all.register(torch.Tensor)
+@fns.all.register(torch.Tensor)
 def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> Union[torch.Tensor, bool]:
     if axis is None:
         return torch.all(a)
     return torch.all(a, dim=axis)
 
 
-@functions.allclose.register(torch.Tensor)
+@fns.allclose.register(torch.Tensor)
 def _(a: torch.Tensor, b: torch.Tensor, rtol: float = 1e-05, atol: float = 1e-08, equal_nan: bool = False) -> bool:
     return torch.allclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan)
 
 
-@functions.any.register(torch.Tensor)
+@fns.any.register(torch.Tensor)
 def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> Union[torch.Tensor, bool]:
     if axis is None:
         return torch.any(a)
     return torch.any(a, dim=axis)
 
 
-@functions.count_nonzero.register(torch.Tensor)
+@fns.count_nonzero.register(torch.Tensor)
 def _(a: torch.Tensor, axis: Optional[Union[int, Tuple[int]]] = None) -> torch.Tensor:
     return torch.count_nonzero(a, dim=axis)
 
 
-@functions.isempty.register(torch.Tensor)
+@fns.isempty.register(torch.Tensor)
 def _(a: torch.Tensor) -> bool:
     return a.numel() == 0
 
 
-@functions.isclose.register(torch.Tensor)
+@fns.isclose.register(torch.Tensor)
 def _(a: torch.Tensor, b: torch.Tensor, rtol: float = 1e-05, atol: float = 1e-08, equal_nan: bool = False):
     return torch.isclose(a, b, atol=atol, rtol=rtol, equal_nan=equal_nan)
 
 
-@functions.maximum.register(torch.Tensor)
+@fns.maximum.register(torch.Tensor)
 def _(x1: torch.Tensor, x2: torch.Tensor) -> torch.Tensor:
     if not isinstance(x2, torch.Tensor):
         x2 = torch.tensor(x2, device=x1.data.device)
     return torch.maximum(x1, x2)
 
 
-@functions.minimum.register(torch.Tensor)
+@fns.minimum.register(torch.Tensor)
 def _(x1: torch.Tensor, x2: torch.Tensor) -> torch.Tensor:
     if not isinstance(x2, torch.Tensor):
         x2 = torch.tensor(x2, device=x1.data.device)
     return torch.minimum(x1, x2)
 
 
-@functions.ones_like.register(torch.Tensor)
+@fns.ones_like.register(torch.Tensor)
 def _(a: torch.Tensor) -> torch.Tensor:
     return torch.ones_like(a)
 
 
-@functions.where.register(torch.Tensor)
+@fns.where.register(torch.Tensor)
 def _(
     condition: torch.Tensor, x: Union[torch.Tensor, float, bool], y: Union[torch.Tensor, float, bool]
 ) -> torch.Tensor:
     return torch.where(condition, x, y)
 
 
-@functions.zeros_like.register(torch.Tensor)
+@fns.zeros_like.register(torch.Tensor)
 def _(a: torch.Tensor) -> torch.Tensor:
     return torch.zeros_like(a)
+
+
+@fns.stack.register(torch.Tensor)
+def _(x: List[torch.Tensor], axis: int = 0) -> List[torch.Tensor]:
```

Comment:
The type passed to `.register` will now be different from the type of the first arg, which is a new precedent among these functions. You might want to add a comment in code about that, or an explanation in the README.md

Reply:
Dont think that need it to readme.md until we encounter more precedents like this, for one function doc string and type hints looks enough.

Reply:
There will be at least [three more](https://github.com/openvinotoolkit/nncf/pull/2120/files#diff-7746d0efc38f06575ddc5ce118d41c5e80850aa03a470ef2adb6954f87a57c15R212) "precedents" per backend. And even one is enough for this to have to be documented.

Reply:
Added to section "Add new method or function" and added helper function `_dispatch_list`